lib/Db/ColumnMapper.php

```diff
@@ -44,4 +44,34 @@ public function findAllByTable(int $tableID): array {
 			->where($qb->expr()->eq('table_id', $qb->createNamedParameter($tableID)));
 		return $this->findEntities($qb);
 	}
+
+	/**
+	 * @param array $neededColumnIds
+	 * @return array<string> Array with key = columnId and value = [column-type]-[column-subtype]
+	 * @throws Exception
+	 */
+	public function getColumnTypes(array $neededColumnIds): array {
+		$qb = $this->db->getQueryBuilder();
+		$qb->select('id', 'type', 'subtype')
+			->from($this->table)
+			->where('id IN (:columnIds)')
+			->setParameter('columnIds', $neededColumnIds, IQueryBuilder::PARAM_INT_ARRAY);
+
+		$out = [
+			-1 => 'number',
```

Comment:
What are those values used for? For this a comment would be nice or maybe even not use magic values here but define them in proper consts and make use of them?